server/rooms.ts

```diff
@@ -1363,17 +1366,51 @@ export class GlobalRoomState {
 			// kill server in 10 seconds if it's still set to
 			setTimeout(() => {
 				if (Config.autolockdown && Rooms.global.lockdown === true) {
-					// finally kill the server
-					process.exit();
+					// finally save state / kill the server
+					void this.kill();
 				} else {
 					this.notifyRooms(
 						notifyPlaces,
-						`|html|<div class="broadcsat-red"><b>Automatic server lockdown kill canceled.</b><br /><br />In the last final seconds, the automatic lockdown was manually disabled.</div>`
+						`|html|<div class="broadcast-red"><b>Automatic server lockdown kill canceled.</b><br /><br />In the last final seconds, the automatic lockdown was manually disabled.</div>`
 					);
 				}
 			}, 10 * 1000);
 		}
 	}
+	async kill() {
+		if (Config.persistbattles) await this.writeBattleState();
+		return process.exit();
+	}
+	async writeBattleState() {
+		const buffer: AnyObject = {};
+		const promises = [];
+		for (const [id, room] of Rooms.rooms) {
+			if (!room.battle) continue;
+			const formatid = room.battle.format;
+			if (!buffer[formatid]) buffer[formatid] = {};
+			const promise = room.battle.getLog().then((log) => {
+				buffer[formatid][id] = {
+					inputLog: log.map(item => item.replace(/\r/g, '')).join('\n'),
+					title: room.title,
```

Comment:
Hm, you'll need to extract the players from the roomlog, anyway. The title is probably superfluous, given this?

Reply:
How does `/importinputlog` work? Maybe we should work on that first.

Reply:
Importinputlog requires you to re-add players in the new battle with /addplayer.
The title is also, likewise `Player 1 vs Player 2.` so storing the title, i felt, would be useful. 
I can, however, remove it.

Reply:
Hm, I think it would be useful to actually extract username and format from the inputlog.